src/v/cluster/health_monitor_backend.cc

```diff
@@ -326,13 +305,22 @@ health_monitor_backend::refresh_cluster_health_cache(force_refresh force) {
         co_return errc::success;
     }
 
-    vlog(clusterlog.debug, "refreshing health cache, leader id: {}", leader_id);
+    vlog(
+      clusterlog.debug,
+      "refreshing health cache, leader id: {} self: {}",
+      leader_id,
+      _raft0->self().id());
 
     _refresh_request = ss::make_lw_shared<abortable_refresh_request>(
       *leader_id, std::move(holder), std ::move(units));
 
-    co_return co_await _refresh_request->abortable_await(
-      dispatch_refresh_cluster_health_request(*leader_id));
+    // we either collect the cluster health reports while on raft0 leader or ask
+    // current leader for cluster health
+    auto f = leader_id == _raft0->self().id()
+               ? collect_cluster_health()
+               : dispatch_refresh_cluster_health_request(*leader_id);
```

Comment:
I guess this means that the health report on a follower can be twice as stale as `max_metadata_age` but I guess that's okay (and this was the case earlier anyway)